manifests/init.pp

```diff
@@ -463,6 +463,14 @@
     } ~> Class['systemd::systemctl::daemon_reload'] ~> Service['snmpd']
   }
 
+  # Ubuntu 16.04+ use systemd
+  if ( $facts['os']['name'] == 'Ubuntu' and versioncmp($facts['os']['release']['major'], '16.04') >= 0 ) {
```

Comment:
instead of that if, should we check what the service_provider fact says?

Reply:
hey now thats an idea!